### PR TITLE
Implement host compute metadata validation tests

### PR DESCRIPTION
**Pull request by jamesnohilly**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Implement host compute metadata validation during `ComputeOnHost` translation tests.

#### Which issue(s) this PR is related to:

https://github.com/torch-spyre/torch-spyre/issues/1567

#### Does this PR introduce a user-facing change?

No

### CODE REVIEW

File: tests/test_prepare_kernel.py [hunk 1]
```diff
@@ -138,6 +153,177 @@ def test_job_plan_step_index_out_of_range(self):
...
+
+            # Should succeed without exceptions
+            job_plan = torch_spyre._C.prepare_kernel(spyrecode_dir)
+
+            # Verify JobPlan was created
+            assert job_plan is not None
+            assert isinstance(job_plan, torch_spyre._C.JobPlan)
+
+            # Verify it has at least one step
+            assert job_plan.num_steps() >= 1
```
Comment (JRosenkranz) [inline tests/test_prepare_kernel.py]: nit: This could be `job_plan.num_steps() == 1`

File: tests/test_prepare_kernel.py [hunk 2]
```diff
@@ -138,6 +153,177 @@ def test_job_plan_step_index_out_of_range(self):
...
+            }
+            spyrecode_dir = self.create_mock_spyrecode(
+                tmpdir, exec_command="ComputeOnHost", exec_properties=properties
+            )
+
+            with pytest.raises(
+                RuntimeError, match="ComputeOnHost command missing 'ishape' property"
+            ):
+                torch_spyre._C.prepare_kernel(spyrecode_dir)
+
```
Comment (JRosenkranz) [inline tests/test_prepare_kernel.py:261]: nit: We should add a test with an invalid ihandle (ihandle is non-empty, but the named buffer was never created)

**Review by JRosenkranz — DISMISSED** [review]
lgtm

File: tests/test_prepare_kernel.py (hunk 2 above)
Comment (jamesnohilly) [reply tests/test_prepare_kernel.py:261]: I've added a new test for this case in the new commit. Thanks!

**Review by JRosenkranz — DISMISSED** [review]
lgtm

**Comment by jamesnohilly** [comment]
It seems like the step ordering validation broke one of the tests, I'll push a change to fix this.

**Comment by jamesnohilly** [comment]
@JRosenkranz I have now fixed the failure in the tests by adding the additional required steps.

**Review by JRosenkranz — APPROVED** [review]
lgtm
